src/boot/efi/shim.c

```diff
@@ -97,3 +107,9 @@ EFI_STATUS shim_load_image(EFI_HANDLE parent, const EFI_DEVICE_PATH *device_path
 
         return ret;
 }
+
+void shim_retain_protocol(void) {
+        uint8_t buffer = 1;
+
+        (void) efivar_set_raw(MAKE_GUID_PTR(SHIM_LOCK), u"SHIM_RETAIN_PROTOCOL", &buffer, 1, 0);
+}
```

Comment:
Requires a corresponding change in shim ofc, I've asked the maintainers if they are ok with this approach. Seems to work fine: https://github.com/bluca/shim/commit/eb13779fd1b59374cd767c3d902124a7297806f6

Reply:
> Also, `efivar_set_raw` forces the variable to be runtime-access, but userspace has no use for this var.

Consensus from shim maintainers seems to be that it's preferable to have it that way, so that at runtime you can tell that it was used.

Reply:
https://github.com/rhboot/shim/pull/565